drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/domelements/ScenarioCellTextAreaDOMElement.java

```diff
@@ -115,7 +115,6 @@ public void setTabIndex(final int index) {
     public void flush(final String value) {
         String actualValue = value != null && value.isEmpty() ? null : value;
         if (scenarioGridCell != null) {
-            scenarioGridCell.setEditingMode(false);
```

Comment:
@jomarko 
Are you sure about removing that line? I remember times ago I put it there for some reason, not sure if it is still needed; anyway, the point at that time was that if ScenarioGridCell.editingMode is "true", the editing does not start. When you press "enter" or anyway end editing, the "flush" method gets invoked, and after that the cell must be editable again.

Reply:
Yes I am sure, see https://github.com/kiegroup/appformer/pull/678/files, detach is always called when flush.

Reply:
OK

Reply:
@jomarko 
I've read your modification to ScenarioCellTextAreaDOMElement and BaseSingletonDOMElementFactory, what I see is
1) it is "true" that BaseSingletonDOMElementFactory.flush calls detach that in turns set editing mode false
2) it is not true that ScenarioCellTextAreaDOMElement.flush calls detach; so whatever code calls directly ScenarioCellTextAreaDOMElement.flush (being public is perfectly valid) does not reset the editing mode
3) moving setEditingMode(false) to the detach method add an unneeded "indirection" and contrived logic, binding together an event (detach) and a behavior (set editingmode) that are not strictly related
4) the editingmode is first of all a *concept*, that it is related to the fact that the user has ended editing; and "ending editing" is represented by the "flush" method, not by "detach"
5) it is just by "chance" that detach and flush are somehow related in our current situation.